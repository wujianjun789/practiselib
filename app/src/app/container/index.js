/**
 * Created by a on 2017/6/29.
 */
import React, { Component } from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux'
import PropTypes from 'prop-types';
import '../../../public/styles/app.less';
import Card from './Card';
import UserCenter from '../../common/containers/UserCenter';
import Overlayer from '../../common/containers/Overlayer';
import {loginHandler} from '../../api/login'
import {getModule} from '../action/index'
/**
 * @param {String} title required
 * @param {String} name  required
 * @param {Array}  items required
 */
export class App extends Component{
    constructor(props){
        super(props);
        this.loginFail = this.loginFail.bind(this);
        this.loginSuccess = this.loginSuccess.bind(this);
    }

    componentWillMount(){
        console.log(localStorage)
        /*if(sessionStorage.sessionID==0||sessionStorage.sessionID==null||sessionStorage.sessionID==""){
            
            this.props.router.push('/login')
            
        }else{

            this.props.actions.loginHandler(sessionStorage.username, sessionStorage.password, this.loginSuccess, this.loginFail);

        }*/
        // if(this.props.userCenter.islogin!=1){
        //     this.props.router.push('/login')
        // }
        const {actions} = this.props;
        actions && actions.getModule();
    }

    componentDidMount() {
    }

    componentWillReceiveProps(nextProps) {



    }

    loginFail(){
        this.setState({style:{visibility: 'visible'}})
    }

    loginSuccess(){
        this.props.router.push('/')
    }

    render(){
        const {title="StarRiver", name="智慧路灯管理系统", items} = this.props;
        return (
            <div className="app">
                <div className="header clearfix">
                    <div className="header-left clearfix">
                        <span className="icon icon-logo"></span>
                        <span className="tit">{title}</span>
                    </div>
                    <UserCenter router={this.props.router}/>
                    <div className="header-right clearfix">
                        <span className="name">{name}</span>
                    </div>
                </div>
                <div className="cont">
                    <ul className="clearfix">
                        {items.map((item, index)=><li key={item.key}><Card _key={item.key} title={item.title} link={item.link}/></li>)}
                    </ul>
                </div>
                <Overlayer />
            </div>
        )
    }
}

const mapStateToProps = (state, ownProps) => {
    let app = state.app;
    return {
        title: app.title,
        name: app.name,
        items: app.items,
        userCenter:state.userCenter
    }
}

const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        actions: bindActionCreators({
            getModule: getModule,
            loginHandler
        }, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(App);

App.propTypes = {
    title: PropTypes.string,
    name: PropTypes.string,
    items: PropTypes.arrayOf(
        PropTypes.shape({
            key: PropTypes.string.isRequired,
            title: PropTypes.string.isRequired,
            link: PropTypes.string.isRequired
        })
    ).isRequired
}