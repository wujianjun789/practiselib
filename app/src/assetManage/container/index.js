/**
 * Created by a on 2017/7/17.
 */
import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import {injectIntl} from 'react-intl';

import '../../../public/styles/assetManage-model.less';
import '../../../public/styles/assetManage-statistics.less';

import HeadBar from '../../components/HeadBar';
import SideBar from '../../components/SideBar';
import Overlayer from '../../common/containers/Overlayer';

import {getModelData, TreeData} from '../../data/assetModels';
import {treeViewInit, onToggleById} from '../../common/actions/treeView';

import {sideBarToggled} from '../action/index';
import {treeViewNavigator} from '../../common/util/index';
class AssetManageIndex extends Component {
  constructor(props) {
    super(props);

    this.state = {
      pathname: ""
    }

    this.initModel = this.initModel.bind(this);
    this.initTreeData = this.initTreeData.bind(this);
    this.onToggle = this.onToggle.bind(this);

    this.getMathNode = this.getMathNode.bind(this);
  }

  componentWillMount() {
    this.mounted = true;
    this.initModel();
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  componentWillReceiveProps(nextProps){
    if(nextProps.location.pathname !==  this.props.location.pathname){
      this.props.actions.onToggleById(nextProps.location.pathname);
    }
  }
  componentDidUpdate(){
    // console.log(this.props);
    // const {location} = this.props;
    // if(location.pathname !== this.state.pathname){
    //   this.setState({pathname: location.pathname}, ()=>{
    //     this.initModel();
    //   });
    // }
  }
  initModel(){
    getModelData(() => {this.mounted && this.initTreeData();});
  }

  initTreeData() {
    TreeData.map(tree=>{
      return Object.assign({}, tree, {name:this.props.intl.formatMessage({id:tree.name})});
    })
    this.props.actions.treeViewInit(TreeData);
    treeViewNavigator(TreeData, this.props.router);

    const node = this.getMathNode(TreeData, this.props.location.pathname);
    if(node){
      this.onToggle(node);
    }
  }

  getMathNode(data, pathname){
    for(let i=0;i<data.length;i++){
      if(!data[i].children && data[i].link.indexOf(pathname) > -1){
        return data[i];
      }else{
        if(data[i].children){
          const node = this.getMathNode(data[i].children, pathname);
          if(node){
            return node;
          }
        }

      }
    }

    return null;
  }

  onToggle(node) {
    const {actions} = this.props;
    actions.sideBarToggled(node);
  }

  render() {
    let parentPath = '';
    let childPath = '';
    const {routes} = this.props;
    if (routes.length > 4) {
      parentPath = routes[4].path;
    }

    if (routes.length > 5) {
      childPath = routes[5].path;
    }

    return <div className={'container ' + 'asset-' + parentPath + ' ' + parentPath + '-' + childPath}>
      <HeadBar moduleName="app.asset.manage" router={this.props.router}/>
      <SideBar onToggle={this.onToggle}/>
      {this.props.children}
      <Overlayer />
    </div>;
  }
}

function mapStateToProps(state) {
  return {
    userCenter:state.userCenter,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      treeViewInit: treeViewInit,
      onToggleById: onToggleById,
      sideBarToggled: sideBarToggled,
    }, dispatch),
  };
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(injectIntl(AssetManageIndex));