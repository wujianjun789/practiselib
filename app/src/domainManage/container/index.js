/**
 * Created by a on 2017/7/17.
 */
import React,{Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import HeadBar from '../../components/HeadBar'
import SideBar from '../../components/SideBar'
import Overlayer from '../../common/containers/Overlayer'

import {TreeData} from '.././../data/domainModel'
import {treeViewInit} from '../../common/actions/treeView'
import {sideBarToggled} from '../action/index'
class DomainManageIndex extends Component{
    constructor(props){
        super(props);
        this.initTreeData = this.initTreeData.bind(this);
        this.onToggle = this.onToggle.bind(this);
    }

    componentWillMount(){
        if(sessionStorage.sessionID==0||sessionStorage.sessionID==null||sessionStorage.sessionID==""){
            this.props.router.push('/login') 
        }else{}
        this.initTreeData();
    }

    componentWillUnmount(){

    }

    componentDidMount(){
        console.log(this.props.userCenter); 
    }

    initTreeData(){
        this.props.actions.treeViewInit(TreeData);
    }

    onToggle(node){
        this.props.actions.sideBarToggled(node);
    }

    render(){
        let path = "";
        const {children} = this.props;
        if(children){
            path = children.props.route.path;
        }
        return <div className={"container "+"domain-"+path}>
            <HeadBar moduleName={/*path=="domainEdit"?*/"域管理"/*:"地图预览"*/} router={this.props.router}/>
            <SideBar onToggle={this.onToggle}/>
            {this.props.children}
            <Overlayer />
        </div>
    }
}

function mapStateToProps(state) {
    return {
        userCenter:state.userCenter
    }
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            treeViewInit: treeViewInit,
            sideBarToggled: sideBarToggled
        }, dispatch)
    }
}
export default connect(
    mapStateToProps,
    mapDispatchToProps
)(DomainManageIndex);