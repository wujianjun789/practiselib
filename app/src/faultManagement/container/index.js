import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import Table from '../../components/Table';
import Page from '../../components/Page';
import SideBarInfo from '../../components/SideBarInfo';
import Select from '../../components/Select.1';
import ConfirmPopup from '../../components/ConfirmPopup';
import Immutable from 'immutable';
import { overlayerShow, overlayerHide } from '../../common/actions/overlayer';

import Content from '../../components/Content';
import { treeViewInit } from '../../common/actions/treeView';
import { injectIntl } from 'react-intl';
import { getChildDomainList } from '../../api/domain';
import { DatePicker } from 'antd';
import moment from 'moment';
import PieChart from '../../lightManage/utils/pieChart';
import { spliceInArray } from '../../util/algorithm';
import { getFaultOrAlertList } from '../../api/asset';

export class Fault extends Component {
  constructor(props) {
    super(props);
    this.state = {
      model: '',
      collapse: false,
      infoCollapse: false,
      page: Immutable.fromJS({
        pageSize: 15,
        current: 1,
        total: 0
      }),
      data: Immutable.fromJS([]),
      domainList: {
        titleField: 'name',
        valueField: 'name',
        index: 0,
        value: '',
        options: []
      },
      levelList: {
        titleField: 'title',
        valueField: 'name',
        index: 0,
        value: 'deadly',
        options: [
          {
            title: '致命',
            name: 'deadly'
          },
          {
            title: '严重',
            name: 'seriousness'
          },
          {
            title: '一般',
            name: 'ordinary'
          },
          {
            title: '提示',
            name: 'energy'
          }
        ]
      },
      start: moment(),
      end: moment(),
      statisticalInfo: {
        deadly: 50,
        serious: 30,
        general: 10,
        hint: 10
      }
    };

    this.columns = [
      {
        id: 0,
        field: 'model',
        title: this.props.intl.formatMessage({
          id: 'sysOperation.alarm.device.model'
        })
      },
      {
        id: 1,
        field: 'name',
        title: this.props.intl.formatMessage({ id: 'app.device.name' })
      },
      {
        id: 3,
        field: 'param',
        title: this.props.intl.formatMessage({ id: 'sysOperation.alarm.param' })
      },
      {
        id: 3,
        field: 'level',
        title: this.props.intl.formatMessage({ id: 'sysOperation.alarm.level' })
      },
      {
        id: 4,
        field: 'threshold',
        title: this.props.intl.formatMessage({
          id: 'sysOperation.alarm.threshold'
        })
      },
      {
        id: 5,
        field: 'value',
        title: this.props.intl.formatMessage({
          id: 'sysOperation.alarm.test.value'
        })
      },
      {
        id: 6,
        field: 'time',
        title: this.props.intl.formatMessage({ id: 'sysOperation.fault.time' })
      }
    ];
  }

  componentWillMount() {
    this.mounted = true;
    this.setState({ model: this.props.params.fault }, this.requestSearch);
    getChildDomainList(data => {
      this.mounted && this.initDomainList(data);
    });
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  componentDidUpdate() {
    const { sidebarNode } = this.props;
    if (
      sidebarNode &&
      !sidebarNode.children &&
      sidebarNode.id != this.state.model
    ) {
      this.setState({ model: sidebarNode.id }, this.requestSearch);
    }
  }

  formatIntl = formatId => {
    const { intl } = this.props;
    return intl ? intl.formatMessage({ id: formatId }) : null;
  };

  initDomainList = data => {
    let domainList = Object.assign(
      {},
      this.state.domainList,
      { index: 0 },
      { value: data.length ? data[0].name : '' },
      { options: data }
    );
    this.setState({ domainList: domainList }, this.requestSearch);
  };

  requestSearch = () => {
    let model = '';
    switch (this.state.model) {
      case 'alarm':
        model = 'alerts';
        break;
      case 'fault':
        model = 'faults';
        break;
    }
    const { domainList, levelList, start, end, page } = this.state;
    const { pageSize, current } = page.toJS();
    if (domainList.options.length) {
      const domainId = domainList.options[domainList.index]['id'];
      // const level = levelList.options[levelList.index]['name'];
      const level = levelList.index;
      const offset = pageSize * (current - 1);
      const limit = pageSize;
      getFaultOrAlertList(
        model,
        domainId,
        level,
        start,
        end,
        offset,
        limit,
        res => {
          this.setState({ data: res });
        }
      );
    }
  };

  initPageSize = data => {
    let page = this.state.page.set('total', data.count);
    this.setState({
      page: page
    });
  };

  pageChange = (current, pageSize) => {
    let page = this.state.page.set('current', current);
    this.setState(
      {
        page: page
      },
      () => {
        this.requestSearch();
      }
    );
  };

  collapseHandler = id => {
    this.setState({
      [id]: !this.state[id]
    });
  };

  selectChange = (event, key) => {
    let index = event.target.selectedIndex;
    let list = this.state[key];
    list.index = index;
    list.value = list.options[index].name;
    this.setState({ [key]: list }, () => {
      this.requestSearch();
    });
  };

  dateChange = (id, value) => {
    this.setState({ [id]: value }, () => {
      const { start, end } = this.state;
      let prompt = start && end && end.isBefore(start);
      this.setState({
        prompt: Object.assign({}, this.state.prompt, { date: prompt })
      });
    });
  };

  drawChart = ref => {
    if (ref == null) {
      this.piechart.destroy();
    } else {
      const { statisticalInfo } = this.state;
      let data = [
        statisticalInfo.deadly,
        statisticalInfo.serious,
        statisticalInfo.general,
        statisticalInfo.hint
      ];
      this.piechart = new PieChart({
        wrapper: ref,
        data: data,
        color: ['#F83D59', '#FA919C', '#FA9E17', '#FBEF35']
      });
    }
  };

  render() {
    const {
      collapse,
      infoCollapse,
      page,
      data,
      domainList,
      typeList,
      levelList,
      start,
      end,
      statisticalInfo,
      allChecked,
      checked
    } = this.state;
    return (
      <Content className={'offset-right ' + (collapse ? 'collapsed' : '')}>
        <div className="heading">
          <Select
            id="domain"
            titleField={domainList.titleField}
            valueField={domainList.valueField}
            options={domainList.options}
            value={domainList.value}
            onChange={e => {
              this.selectChange(e, 'domainList');
            }}
          />
          {/* <Select id="type" titleField={typeList.titleField} valueField={typeList.valueField}
          options={typeList.options} value={typeList.value} onChange={(e) => {this.selectChange(e, 'typeList');}}/> */}
          <Select
            id="level"
            titleField={levelList.titleField}
            valueField={levelList.valueField}
            options={levelList.options}
            value={levelList.value}
            onChange={e => {
              this.selectChange(e, 'levelList');
            }}
          />
          <div className="datePicker">
            <DatePicker
              id="startDate"
              format="YYYY/MM/DD"
              placeholder="开始日期"
              style={{ width: '106px' }}
              defaultValue={start}
              value={start}
              onChange={value => this.dateChange('start', value)}
            />
            <span>-</span>
            <DatePicker
              id="endDate"
              format="YYYY/MM/DD"
              placeholder="结束日期"
              style={{ width: '106px' }}
              defaultValue={end}
              value={end}
              onChange={value => this.dateChange('end', value)}
            />
          </div>
        </div>
        <div className="table-container">
          <Table columns={this.columns} data={data} />
          <Page
            className={'page ' + (page.get('total') == 0 ? 'hidden' : '')}
            pageSize={page.get('pageSize')}
            current={page.get('current')}
            total={page.get('total')}
            onChange={this.pageChange}
          />
        </div>
        <SideBarInfo
          collapseHandler={this.collapseHandler}
          className={infoCollapse ? 'infoCollapse ' : ''}
        >
          <div className="panel panel-default statistical-info">
            <div
              className="panel-heading"
              role="presentation"
              onClick={() => {
                !collapse && this.collapseHandler('infoCollapse');
              }}
            >
              <span className="icon_chart" />
              {this.props.intl.formatMessage({
                id: 'sysOperation.statistical.info'
              })}
              <span className="icon icon_collapse pull-right" />
            </div>
            <div className={'panel-body' + (infoCollapse ? 'collapsed' : '')}>
              <div className="left">
                <div id="alarm" className="circle" ref={this.drawChart} />
              </div>
              <div className="right">
                <div className="count deadly">
                  <div className="dot" />
                  {this.formatIntl('sysOperation.count.deadly')} :{' '}
                  {statisticalInfo.deadly
                    ? statisticalInfo.deadly
                    : this.formatIntl('sysOperation.noCount')}
                </div>
                <div className="count serious">
                  <div className="dot" />
                  {this.formatIntl('sysOperation.count.serious')} :{' '}
                  {statisticalInfo.serious
                    ? statisticalInfo.serious
                    : this.formatIntl('sysOperation.noCount')}
                </div>
                <div className="count general">
                  <div className="dot" />
                  {this.formatIntl('sysOperation.count.general')} :{' '}
                  {statisticalInfo.general
                    ? statisticalInfo.general
                    : this.formatIntl('sysOperation.noCount')}
                </div>
                <div className="count hint">
                  <div className="dot" />
                  {this.formatIntl('sysOperation.count.hint')} :{' '}
                  {statisticalInfo.hint
                    ? statisticalInfo.hint
                    : this.formatIntl('sysOperation.noCount')}
                </div>
              </div>
            </div>
          </div>
        </SideBarInfo>
      </Content>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    sidebarNode: state.systemOperation.get('sidebarNode')
  };
};

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    {
      treeViewInit,
      overlayerShow,
      overlayerHide
    },
    dispatch
  )
});

export default connect(mapStateToProps, mapDispatchToProps)(injectIntl(Fault));
