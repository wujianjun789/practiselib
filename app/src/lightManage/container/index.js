/**
 * Created by a on 2017/8/24.
 */
import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import '../../../public/styles/lightManage-map.less';
import '../../../public/styles/lightManage-control.less';

import HeadBar from '../../components/HeadBar';
import SideBar from '../../components/SideBar';
import Overlayer from '../../common/containers/Overlayer';

import {getModelData, TreeData} from '.././../data/lightModel';
import {treeViewInit, onToggleById} from '../../common/actions/treeView';
// import {sideBarToggled} from '../action/index'
class lightManageIndex extends Component {
  constructor(props) {
    super(props);
    this.initTreeData = this.initTreeData.bind(this);
    this.onToggle = this.onToggle.bind(this);
  }

  componentWillMount() {
    this.mounted = true;
    getModelData(null, () => {
      this.mounted && this.initTreeData();
    });
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  componentDidMount() {
  }

  componentWillReceiveProps(nextProps){
    if(nextProps.location.pathname !== this.props.location.pathname){
      this.props.actions.onToggleById(nextProps.location.pathname);
    }
  }

  initTreeData() {
    this.props.actions.treeViewInit(TreeData);
  }

  onToggle(node) {
    // console.log(node);
    // this.props.actions.sideBarToggled(node);
  }

  render() {
    let parentPath = '';
    let childPath = '';
    const {routes} = this.props;
    if (routes.length > 4) {
      parentPath = routes[4].path;
    }

    if (routes.length > 5) {
      childPath = routes[5].path;
    }
    return <div className={'container light-' + parentPath + ' ' + parentPath + '-' + childPath}>
      <HeadBar moduleName={'app.light'} router={this.props.router}/>
                    
      <SideBar onToggle={this.onToggle}/>
      {this.props.children}
      <Overlayer />
    </div>;
  }
}

function mapStateToProps(state) {
  return {
    userCenter:state.userCenter,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      treeViewInit: treeViewInit,
      onToggleById: onToggleById
      // sideBarToggled: sideBarToggled
    }, dispatch),
  };
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(lightManageIndex);