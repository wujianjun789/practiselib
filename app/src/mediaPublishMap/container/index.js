/**
 * Created by a on 2017/10/19.
 */
import React,{Component} from 'react';

import Content from '../../components/Content';
import MapView from '../../components/MapView';
export default class MediaPublishMap extends Component{
    constructor(props){
        super(props);
        this.state = {
            map:{
                id: "mediaPublishMap",
                latlng: []
            }
        };
    }

    render(){
        const {map} = this.state;
        return <Content>
            <MapView mapData={map}></MapView>
        </Content>
    }
}