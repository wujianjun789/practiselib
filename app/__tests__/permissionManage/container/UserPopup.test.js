import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import {UserPopup} from '../../../src/permissionManage/container/UserPopup';
// import {Provider} from 'react-redux';
import Immutable from 'immutable';
// import configureStore from '../../../src/store/configureStore';

describe('<UserPopup',()=>{
    // const store = configureStore();
    const classNameAdd = 'user-add-popup';
    const titleAdd = '添加用户';
    const classNameEdit = 'user-edit-popup';
    const titleEdit = '用户资料';
    const data = {
        
    }
    const modules = [
        {key: 'asset', title: '资产管理', link: '/assetManage/manage'},
        {key: 'permission', title: '权限管理', link: '/permissionManage'},
        {key: 'maintenance', title: '系统运维', link: '/systemOperation/lampConCenter'},
        {key: 'control', title: '系统控制', link: '/'},
        {key: 'report', title: '报表管理', link: '/'},
        {key: 'publish', title: '媒体发布', link: '/'},
        {key: 'visual', title: '可视化', link: '/'},
        {key: 'domain', title: '域管理', link: '/domainManage/domainEdit'}
    ]
    it('render normal add',()=>{
        const cmp = shallow(<UserPopup className={classNameAdd} title={titleAdd} modules={modules}/>);
        const container = cmp.find(`.${classNameAdd}`);
        expect(container.length).toBe(1);

        const panel = cmp.find('Panel');
        expect(panel.length).toBe(1);
        expect(panel.prop('title')).toBe(titleAdd);

        const inputCheck = cmp.find('InputCheck');
        expect(inputCheck.length).toBe(5);

        const select = cmp.find('Select');
        expect(select.length).toBe(1);
    })
})