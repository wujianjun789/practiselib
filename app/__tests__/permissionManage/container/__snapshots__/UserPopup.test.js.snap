exports[`<UserPopup add snapshot 1`] = `
<div
  className="modal-content user-add-popup">
  <div
    className="modal-header">
    <button
      aria-label="Close"
      className="close"
      data-dismiss="modal"
      onClick={[Function]}
      type="button">
      <span
        aria-hidden="true">
        ×
      </span>
    </button>
    <h4
      className="modal-title">
      添加用户
    </h4>
  </div>
  <div
    className="modal-body">
    <div
      className="form-group row basic-info">
      <div
        className="inputCheck username">
        <label
          className="col-sm-2 control-label">
          用户名
          :
        </label>
        <div
          className="has-feedback col-sm-4 ">
          <input
            className="form-control"
            disabled={false}
            id="username"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="请输入用户名"
            type="text"
            value="" />
          <span
            aria-hidden="true"
            className="glyphicon  form-control-feedback" />
          <span
            className="reminder">
            
          </span>
        </div>
      </div>
      <label
        className="col-sm-2 control-label">
        用户等级:
      </label>
      <select
        className="select role"
        onChange={[Function]}
        placeholder={undefined}
        value="访客">
        <option
          value="访客">
          访客
        </option>
        <option
          value="设备操作员">
          设备操作员
        </option>
        <option
          value="设备管理员">
          设备管理员
        </option>
        <option
          value="系统管理员">
          系统管理员
        </option>
      </select>
      <div
        className="inputCheck lastName">
        <label
          className="col-sm-2 control-label">
          姓氏
          :
        </label>
        <div
          className="has-feedback col-sm-4 ">
          <input
            className="form-control"
            disabled={false}
            id="lastName"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="请输入姓氏"
            type="text"
            value="" />
          <span
            aria-hidden="true"
            className="glyphicon  form-control-feedback" />
          <span
            className="reminder">
            
          </span>
        </div>
      </div>
      <div
        className="inputCheck firstName">
        <label
          className="col-sm-2 control-label">
          名字
          :
        </label>
        <div
          className="has-feedback col-sm-4 ">
          <input
            className="form-control"
            disabled={false}
            id="firstName"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="请输入名字"
            type="text"
            value="" />
          <span
            aria-hidden="true"
            className="glyphicon  form-control-feedback" />
          <span
            className="reminder">
            
          </span>
        </div>
      </div>
      <div
        className="inputCheck password ">
        <label
          className="col-sm-2 control-label">
          密码
          :
        </label>
        <div
          className="has-feedback col-sm-4 ">
          <input
            className="form-control"
            disabled={false}
            id="password"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="请输入密码"
            type="password"
            value="" />
          <span
            aria-hidden="true"
            className="glyphicon  form-control-feedback" />
          <span
            className="reminder">
            
          </span>
        </div>
      </div>
      <div
        className="inputCheck rePassword ">
        <label
          className="col-sm-2 control-label">
          重复密码
          :
        </label>
        <div
          className="has-feedback col-sm-4 ">
          <input
            className="form-control"
            disabled={false}
            id="rePassword"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="请再次输入密码"
            type="password"
            value="" />
          <span
            aria-hidden="true"
            className="glyphicon  form-control-feedback" />
          <span
            className="reminder">
            
          </span>
        </div>
      </div>
    </div>
    <div
      className="form-group row module-per">
      <label
        className="col-sm-2 control-label">
        模块权限:
      </label>
      <div
        className="col-sm-10">
        <div
          className="row">
          <label
            className="checkbox-inline">
            <input
              defaultChecked={false}
              name="module"
              type="checkbox"
              value="asset" />
             
            资产管理
          </label>
          <label
            className="checkbox-inline">
            <input
              defaultChecked={false}
              name="module"
              type="checkbox"
              value="permission" />
             
            权限管理
          </label>
          <label
            className="checkbox-inline">
            <input
              defaultChecked={false}
              name="module"
              type="checkbox"
              value="maintenance" />
             
            系统运维
          </label>
          <label
            className="checkbox-inline">
            <input
              defaultChecked={false}
              name="module"
              type="checkbox"
              value="control" />
             
            系统控制
          </label>
        </div>
        <div
          className="row">
          <label
            className="checkbox-inline">
            <input
              defaultChecked={false}
              name="module"
              type="checkbox"
              value="report" />
             
            报表管理
          </label>
          <label
            className="checkbox-inline">
            <input
              defaultChecked={false}
              name="module"
              type="checkbox"
              value="publish" />
             
            媒体发布
          </label>
          <label
            className="checkbox-inline">
            <input
              defaultChecked={false}
              name="module"
              type="checkbox"
              value="visual" />
             
            可视化
          </label>
          <label
            className="checkbox-inline">
            <input
              defaultChecked={false}
              name="module"
              type="checkbox"
              value="domain" />
             
            域管理
          </label>
        </div>
        
      </div>
    </div>
  </div>
  <div
    className="modal-footer">
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      取消
    </button>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button">
      确认
    </button>
  </div>
</div>
`;

exports[`<UserPopup edit snapshot 1`] = `
<div
  className="modal-content user-edit-popup">
  <div
    className="modal-header">
    <button
      aria-label="Close"
      className="close"
      data-dismiss="modal"
      onClick={[Function]}
      type="button">
      <span
        aria-hidden="true">
        ×
      </span>
    </button>
    <h4
      className="modal-title">
      用户资料
    </h4>
  </div>
  <div
    className="modal-body">
    <div
      className="form-group row basic-info">
      <div
        className="inputCheck username">
        <label
          className="col-sm-2 control-label">
          用户名
          :
        </label>
        <div
          className="has-feedback col-sm-4 ">
          <input
            className="form-control"
            disabled={false}
            id="username"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="请输入用户名"
            type="text"
            value="" />
          <span
            aria-hidden="true"
            className="glyphicon  form-control-feedback" />
          <span
            className="reminder">
            
          </span>
        </div>
      </div>
      <label
        className="col-sm-2 control-label">
        用户等级:
      </label>
      <select
        className="select role"
        onChange={[Function]}
        placeholder={undefined}
        value="访客">
        <option
          value="访客">
          访客
        </option>
        <option
          value="设备操作员">
          设备操作员
        </option>
        <option
          value="设备管理员">
          设备管理员
        </option>
        <option
          value="系统管理员">
          系统管理员
        </option>
      </select>
      <div
        className="inputCheck lastName">
        <label
          className="col-sm-2 control-label">
          姓氏
          :
        </label>
        <div
          className="has-feedback col-sm-4 ">
          <input
            className="form-control"
            disabled={false}
            id="lastName"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="请输入姓氏"
            type="text"
            value="" />
          <span
            aria-hidden="true"
            className="glyphicon  form-control-feedback" />
          <span
            className="reminder">
            
          </span>
        </div>
      </div>
      <div
        className="inputCheck firstName">
        <label
          className="col-sm-2 control-label">
          名字
          :
        </label>
        <div
          className="has-feedback col-sm-4 ">
          <input
            className="form-control"
            disabled={false}
            id="firstName"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="请输入名字"
            type="text"
            value="" />
          <span
            aria-hidden="true"
            className="glyphicon  form-control-feedback" />
          <span
            className="reminder">
            
          </span>
        </div>
      </div>
      <div
        className="inputCheck password ">
        <label
          className="col-sm-2 control-label">
          密码
          :
        </label>
        <div
          className="has-feedback col-sm-4 ">
          <input
            className="form-control"
            disabled={false}
            id="password"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="请输入密码"
            type="password"
            value="" />
          <span
            aria-hidden="true"
            className="glyphicon  form-control-feedback" />
          <span
            className="reminder">
            
          </span>
        </div>
      </div>
      <div
        className="inputCheck rePassword ">
        <label
          className="col-sm-2 control-label">
          重复密码
          :
        </label>
        <div
          className="has-feedback col-sm-4 ">
          <input
            className="form-control"
            disabled={false}
            id="rePassword"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="请再次输入密码"
            type="password"
            value="" />
          <span
            aria-hidden="true"
            className="glyphicon  form-control-feedback" />
          <span
            className="reminder">
            
          </span>
        </div>
      </div>
    </div>
    <div
      className="form-group row module-per">
      <label
        className="col-sm-2 control-label">
        模块权限:
      </label>
      <div
        className="col-sm-10">
        <div
          className="row">
          <label
            className="checkbox-inline">
            <input
              defaultChecked={false}
              name="module"
              type="checkbox"
              value="asset" />
             
            资产管理
          </label>
          <label
            className="checkbox-inline">
            <input
              defaultChecked={false}
              name="module"
              type="checkbox"
              value="permission" />
             
            权限管理
          </label>
          <label
            className="checkbox-inline">
            <input
              defaultChecked={false}
              name="module"
              type="checkbox"
              value="maintenance" />
             
            系统运维
          </label>
          <label
            className="checkbox-inline">
            <input
              defaultChecked={false}
              name="module"
              type="checkbox"
              value="control" />
             
            系统控制
          </label>
        </div>
        <div
          className="row">
          <label
            className="checkbox-inline">
            <input
              defaultChecked={false}
              name="module"
              type="checkbox"
              value="report" />
             
            报表管理
          </label>
          <label
            className="checkbox-inline">
            <input
              defaultChecked={false}
              name="module"
              type="checkbox"
              value="publish" />
             
            媒体发布
          </label>
          <label
            className="checkbox-inline">
            <input
              defaultChecked={false}
              name="module"
              type="checkbox"
              value="visual" />
             
            可视化
          </label>
          <label
            className="checkbox-inline">
            <input
              defaultChecked={false}
              name="module"
              type="checkbox"
              value="domain" />
             
            域管理
          </label>
        </div>
        
      </div>
    </div>
  </div>
  <div
    className="modal-footer">
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      取消
    </button>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button">
      确认
    </button>
  </div>
</div>
`;
